src/Microsoft.Azure.WebJobs.Host/Singleton/IDistributedLock.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Handle for a lock returned by <see cref="IDistributedLockManager"/>
+    /// </summary>
+    public interface IDistributedLock
+    {
+        /// <summary>
+        /// The Lock identity.  
+        /// </summary>
+        string LockId { get; set; }
+
+        /// <summary>
+        /// Task is set to completed if the lease is lost (after it's been acquired). 
+        /// This allows consumers to get a notification if the lock failed to be renewed. 
+        /// </summary>
+        Task LeaseLost { get; }
```

Comment:
One benefit is that it avoids races around completion.  With an event, you need to take an extra step to register; but if the event fires before you register, you miss it. (This is a lesson from JScript Promises)  

---
In reply to: [120950317](https://github.com/Azure/azure-webjobs-sdk/pull/1184#discussion_r120950317) [](ancestors = 120950317)